<a href="https://colab.research.google.com/github/othyTim/Data-for-ML/blob/main/Data_for_ML_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Decribing Data for ML
Timothy Lingeveldt

###Imports

In [8]:
#Pandas
import pandas as pd


##Car Insurance Data

###Load describe data

In [7]:
#Load data
dataset ='/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week08/Data/Car_Insurance_Claim.csv'
df_insurance = pd.read_csv(dataset)

#Describe data
df_insurance.info()
df_insurance.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 19 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID                   10000 non-null  int64  
 1   AGE                  10000 non-null  object 
 2   GENDER               10000 non-null  object 
 3   RACE                 10000 non-null  object 
 4   DRIVING_EXPERIENCE   10000 non-null  object 
 5   EDUCATION            10000 non-null  object 
 6   INCOME               10000 non-null  object 
 7   CREDIT_SCORE         9018 non-null   float64
 8   VEHICLE_OWNERSHIP    10000 non-null  float64
 9   VEHICLE_YEAR         10000 non-null  object 
 10  MARRIED              10000 non-null  float64
 11  CHILDREN             10000 non-null  float64
 12  POSTAL_CODE          10000 non-null  int64  
 13  ANNUAL_MILEAGE       9043 non-null   float64
 14  VEHICLE_TYPE         10000 non-null  object 
 15  SPEEDING_VIOLATIONS  10000 non-null  

ID    AGE  GENDER      RACE DRIVING_EXPERIENCE    EDUCATION  \
0  569520    65+  female  majority               0-9y  high school   
1  750365  16-25    male  majority               0-9y         none   
2  199901  16-25  female  majority               0-9y  high school   
3  478866  16-25    male  majority               0-9y   university   
4  731664  26-39    male  majority             10-19y         none   

          INCOME  CREDIT_SCORE  VEHICLE_OWNERSHIP VEHICLE_YEAR  MARRIED  \
0    upper class      0.629027                1.0   after 2015      0.0   
1        poverty      0.357757                0.0  before 2015      0.0   
2  working class      0.493146                1.0  before 2015      0.0   
3  working class      0.206013                1.0  before 2015      0.0   
4  working class      0.388366                1.0  before 2015      0.0   

   CHILDREN  POSTAL_CODE  ANNUAL_MILEAGE VEHICLE_TYPE  SPEEDING_VIOLATIONS  \
0       1.0        10238         12000.0        sedan                    0   
1       0.0        10238         16000.0        sedan                    0   
2       0.0        10238         11000.0        sedan                    0   
3       1.0        32765         11000.0        sedan                    0   
4       0.0        32765         12000.0        sedan                    2   

   DUIS  PAST_ACCIDENTS  OUTCOME  
0     0               0      0.0  
1     0               0      1.0  
2     0               0      0.0  
3     0               0      0.0  
4     0               1      1.0

###Source of data

FOund on Kaggle. This insurance data was shared by an insurance company - most of data presented is from legitimate customers, howeversome of the data was generated by the sharer.

###Description of the data

The data has some basic information about the customer such as gender, income, education, driving experience and so forth. it also has information about the vehicle that is being insured such as mileage, vehicle type, yeat model and so forth.

###Target

Y = 'Outcome'

###Row One

Row One presents a person who has taken car insurance as well as the infomation of the car being insured

###Classification Or Regression

Classification as the goal is predict a binery Target

###Features

18 Features

###Row Count

1000

###Challenges do you foresee in cleaning, exploring, or modeling this dataset:

There are a few columns that should be Integer but are object due tp characrers in the fields, such 'age' and 'driving experience'

Irrelevant columns such as 'ID' which will not contribute to the ML models



##Adult Income Dataset

###Load describe data

In [9]:
#Load data
dataset ='/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week08/Data/adult.csv'
df_income = pd.read_csv(dataset)

#Describe data
df_income.info()
df_income.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48842 entries, 0 to 48841
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   age              48842 non-null  int64 
 1   workclass        48842 non-null  object
 2   fnlwgt           48842 non-null  int64 
 3   education        48842 non-null  object
 4   educational-num  48842 non-null  int64 
 5   marital-status   48842 non-null  object
 6   occupation       48842 non-null  object
 7   relationship     48842 non-null  object
 8   race             48842 non-null  object
 9   gender           48842 non-null  object
 10  capital-gain     48842 non-null  int64 
 11  capital-loss     48842 non-null  int64 
 12  hours-per-week   48842 non-null  int64 
 13  native-country   48842 non-null  object
 14  income           48842 non-null  object
dtypes: int64(6), object(9)
memory usage: 5.6+ MB


age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

###Source of data

The data source is from Kaggle and widely used.

###Description of the data

its presents individual's annual income results from various factors such as age, education, gender and so forth.

###Target

Y = 'Income'

###Row One

Row One presents a person and their personal information(attributes that describes the person) such as marital status, race, relationships, education and so forth

###Classification Or Regression

Regression

###Features

14 Features

###Row Count

48842

###Challenges do you foresee in cleaning, exploring, or modeling this dataset:

There are missing data and unique charachers such as '?' as well as prefixes in the target coloumn.
Seems there maybe outliers - 'capital gain' columns
Many of the features are Categorical whic can impact the model run time to scale the data.